src/comm.ts

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+"use strict";
+
+import { DisposableDelegate } from 'phosphor-disposable';
+
+import { 
+  IKernel, IKernelFuture, IKernelMessage, IKernelMessageOptions 
+} from './ikernel';
+
+import { createKernelMessage } from './kernel';
+
+import * as utils from './utils';
+
+
+/**
+ * A client side Comm interface.
+ */
+export
+interface IComm {
+  /**
+   * The uuid for the comm channel.
+   *
+   * Read-only
+   */
+  commId: string;
+
+  /** 
+   * The target name for the comm channel.
+   *
+   * Read-only
+   */
+  targetName: string;
+
+  /**
+   * The onClose handler.
+   */
+  onClose: (data?: any) => void;
+
+  /**
+   * The onMsg handler.
+   */
+  onMsg: (data: any) => void;
+
+  /**
+   * Open a comm with optional data.
+   */
+  open(data?: any, metadata?: any): IKernelFuture;
+
+  /**
+   * Send a comm message to the kernel.
+   */
+  send(data: any, metadata?: any, buffers?:(ArrayBuffer | ArrayBufferView)[]): IKernelFuture;
```

Comment:
what's the expected reply when sending a comm message?

I'm harping on this because I imagine comm messages to be sent *much* more rapidly than normal messages. Every time you move a slider widget, a comm message will be sent. So, I'd like that path to be as efficient as possible.